### PR TITLE
Add RankOne update capability to LU decomposition

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by kortschak** [comment]
LGTM some punctuation.

File: mat64/lu.go [hunk 1]
```diff
@@ -76,8 +76,69 @@ func (lu *LU) Pivot(swaps []int) []int {
 	return swaps
 }
 
+// RankOne updates an LU factorization as if a rank-one update had been applied to
+// the original matrix A, storing the result into the receiver. That is, if in
+// original LU decomposition P * L * U = A, in the updated decomposition
```
Comment (vladimir-ch) [inline mat64/lu.go]: _the_ original LU ... ?

File: mat64/lu.go [hunk 2]
```diff
@@ -76,8 +76,69 @@ func (lu *LU) Pivot(swaps []int) []int {
...
+			xs[i] -= lum.Data[i*lum.Stride+j] * xs[j]
+			tmp := ys[i]
+			ys[i] -= lum.Data[j*lum.Stride+i] * ys[j]
+			lum.Data[i*lum.Stride+j] += beta * xs[i]
+			lum.Data[j*lum.Stride+i] += gamma * tmp
+		}
+	}
+}
+
 // LFrom extracts the lower triangular matrix from an LU factorization.
```
Comment (vladimir-ch) [inline mat64/lu.go]: LFromLU

File: mat64/lu.go [hunk 3]
```diff
@@ -93,7 +154,7 @@ func (t *TriDense) LFrom(lu *LU) {
 }
 
 // UFrom extracts the upper triangular matrix from an LU factorization.
```
Comment (vladimir-ch) [inline mat64/lu.go]: UFromLU

**Pull request merged**